python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -483,20 +483,21 @@ def test_repartitionByRange_dataframe(self):
 
         # test repartitionByRange(numPartitions, *cols)
         df3 = df1.repartitionByRange(2, "name", "age")
-        self.assertEqual(df3.rdd.getNumPartitions(), 2)
-        self.assertEqual(df3.rdd.first(), df2.rdd.first())
-        self.assertEqual(df3.rdd.take(3), df2.rdd.take(3))
+
+        self.assertEqual(df3.select(spark_partition_id()).distinct().count(), 2)
```

Comment:
`df.select(spark_partition_id()).distinct().count()` is not always equivalent to `df.rdd. getNumPartitions()` (e.g. empty partitions, AQE rules like CoalesceShufflePartitions), but in this UT they are the same